lib/Backends/OpenCL/kernels.cl

```diff
@@ -1102,9 +1102,36 @@ __kernel void transposeK_u(__global cl_uint64_t *dest,
 
 __kernel void transpose_uW(__global void *mem, cl_uint32_t dest, cl_uint32_t src,
                          ShapeNHWC odim, ShapeNHWC idim, ShapeNHWC shuffle) {
-  transposeK_u(&mem[dest], &mem[src], odim, idim, shuffle);
+  transpose_uK(&mem[dest], &mem[src], odim, idim, shuffle);
 }
 
+__kernel void transpose_i8K(__global cl_int8_t *dest,
```

Comment:
please avoid code duplication. Transpose of fp32 vs transpose of int8 should not be different. The only difference is a type.

Reply:
so you mean i can write a macro to cover the cases for all these types